ophyd/sim.py

```diff
@@ -344,18 +346,19 @@ class SynGauss(SynSignal):
     """
 
     def __init__(self, name, motor, motor_field, center, Imax, sigma=1,
-                 noise=None, noise_multiplier=1, **kwargs):
+                 noise=None, noise_multiplier=1, random_state=None, **kwargs):
         if noise not in ('poisson', 'uniform', None):
             raise ValueError("noise must be one of 'poisson', 'uniform', None")
         self._motor = motor
-
+        if random_state is None:
+            random_state = np.random
         def func():
             m = motor.read()[motor_field]['value']
             v = Imax * np.exp(-(m - center) ** 2 / (2 * sigma ** 2))
             if noise == 'poisson':
-                v = int(np.random.poisson(np.round(v), 1))
+                v += int(random_state.poisson(np.round(v), 1))
```

Comment:
Actually, the existing code a correct implementation of Poisson noise. You can just rebase and remove the commit 28c64e6 where this was done.

Reply:
👍